page_v1.go

```diff
@@ -79,15 +81,27 @@ func (dp *dataPageReaderV1) init(dDecoder, rDecoder getLevelDecoder, values getV
 	return nil
 }
 
-func (dp *dataPageReaderV1) read(r io.Reader, ph *parquet.PageHeader, codec parquet.CompressionCodec) (err error) {
+func (dp *dataPageReaderV1) read(r io.Reader, sch *schema, ph *parquet.PageHeader, codec parquet.CompressionCodec) (err error) {
 	if ph.DataPageHeader == nil {
 		return errors.Errorf("null DataPageHeader in %+v", ph)
 	}
 
 	if dp.valuesCount = ph.DataPageHeader.NumValues; dp.valuesCount < 0 {
 		return errors.Errorf("negative NumValues in DATA_PAGE: %d", dp.valuesCount)
 	}
-	reader, err := createDataReader(r, codec, ph.GetCompressedPageSize(), ph.GetUncompressedPageSize())
```

Comment:
This change is repeated often:
1. remove call to createDataReader
2. call ReadAll
3. check CRC
4. call newBlockReader

what is the argument against keeping createDataReader and adding the CRC check there?

Reply:
I moved the `ReadAll` and the CRC check to a function `readPageBlock` which combines this sequence which appears 3 times throughout the library. I couldn't integrate `newBlockReader` because the page data is being used differently in `page_v2.go`.